storage/core/src/main/java/com/hortonworks/registries/storage/impl/jdbc/config/HikariConfigFactory.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2016 Hortonworks.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package com.hortonworks.registries.storage.impl.jdbc.config;
+
+import com.hortonworks.registries.storage.common.DatabaseType;
+import com.hortonworks.registries.storage.common.util.Constants;
+import com.hortonworks.registries.storage.common.util.DataSourceUtil;
+import com.zaxxer.hikari.HikariConfig;
+
+import javax.sql.DataSource;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class HikariConfigFactory {
+
+    public static HikariConfig get(DatabaseType type, Map<String, Object> dbProperties) {
+
+        switch (type) {
+            case MYSQL:
+                return mysqlConfig(dbProperties);
+            case POSTGRESQL:
+                return postgresConfig(dbProperties);
+            case ORACLE:
+                return oracleConfig(dbProperties);
+            default:
+                throw new IllegalArgumentException("Unsupported storage provider type: " + type);
+        }
+    }
+
+    private static HikariConfig mysqlConfig(Map<String, Object> dbProperties) {
+        Properties hikariProperties = new Properties();
+        hikariProperties.putAll(dbProperties);
+        return new HikariConfig(hikariProperties);
+    }
+
+    private static HikariConfig postgresConfig(Map<String, Object> props) {
+        return mysqlConfig(props);
+    }
+
+    private static HikariConfig oracleConfig(Map<String, Object> dbProperties) {
+        Map<String, Object> propsCopy = new HashMap<>(dbProperties);
+        propsCopy.remove(Constants.DataSource.CONNECTION_PROPERTY);
+        HikariConfig hikariConfig = mysqlConfig(propsCopy);
```

Comment:
Can't any other database like postgresql be TLS enabled? If so, can this code be applicable to it too?

Reply:
@raju-saravanan can you answer?

Reply:
Other databases can be TLS enabled too. But I haven't explored it for MySQL and Postgres. Since the logic here is to invoke setConnectionProperties function which is specific to Oracle Datasource its safe to assume that this code needs to be extended to enable TLS for other databases apart from Oracle